pandas/core/indexes/datetimes.py

```diff
@@ -321,11 +321,10 @@ def _simple_new(cls, values, name=None, freq=None, tz=None, dtype=None):
 
         dtarr = DatetimeArray._simple_new(values, freq=freq, tz=tz)
         result = object.__new__(cls)
-        result._data = dtarr._data
-        result._freq = dtarr.freq
-        result._tz = dtarr.tz
+        result._eadata = dtarr
         result.name = name
         # For groupby perf. See note in indexes/base about _index_data
+        # TODO: make sure this is updated correctly if edited
```

Comment:
This TODO is for _index_data? In theory that shouldn't happen, since DatetimeIndex is immutable.

Reply:
In _libs.reduction there is a line:

```
object.__setattr__(cached_ityp, '_index_data', islider.buf)
```

which makes me wary.  Is this just never relevant for DTI?

Reply:
That's the intent of all this, to directly mutate the buffer in place. The `.reset` method undoes all this stuff. Nobody else should be messing with it.

Reply:
@TomAugspurger I think all your other comments have been addressed, not sure about this one.  Should this TODO comment be removed?  Some other action taken?

Reply:
Yeah this todo isn't necessary AFAICT.

Also, I really don't think we should have inverted the relationship between the eadata and data attributes.

Reply:
Just because that change isn't going to last through #24024, so I think it was unnecessary.

Reply:
> Also, I really don't think we should have inverted the relationship between the eadata and data attributes. [...] so I think it was unnecessary.

We can't have both 1) _eadata being a property that depends on self.freq and 2) freq be a property that depends on _eadata.

Definitely agree that 24024 should return things to the old pattern.